### ISSUE TITLE
Interpreter escape for %

### ISSUE THREAD

**Issue by elliottinvent** [MODLanguage/grammar#43]
We can use percents like this:
```
key=%letters
```
->
```
{
  "key": "%letters"
}
```
If the reference exists it'll replace the reference:
```
_letters=abc;
key=%letters
```
->
```
{
  "key": "abc"
}
```
However, it's impossible to print the characters "%letters" if `_letters`exists as a reference.

So we need an escape here, and I think it needs to be an interpreter escape:
```
_letters=abc;
key="~%letters"
```
Or:
```
_letters=abc;
key="\%letters"
```
->
```
{
    "key": "%letters"
}
```

**Comment by twalmsley** [comment]
Fixed for Java interpreter.
Ruby to follow soon.

**Comment by twalmsley** [comment]
Ruby fix complete and will be published in v0.3.14
